plugins/crd/plugin_impl_crd.go

```diff
@@ -113,8 +113,40 @@ func (plugin *Plugin) Init() error {
 	return nil
 }
 
+func networkServiceValidation() *apiextv1beta1.CustomResourceValidation {
```

Comment:
I like this approach, it's pretty clean. I think we'd need separate validator functions per-CRD, so one for NetworkService as you have below, and another for NetworkServiceChannel and NetworkServiceEndpoint.

Reply:
Right, that is the plan, each type of CRD will have each own validation since certain fields are different.